src/app/main/main.component.ts

```diff
@@ -166,6 +163,18 @@ export class MainComponent implements OnInit, OnDestroy {
     this.updateStyles();
   }
 
+  readUrlParams() {
+    this.project = this._route.snapshot.paramMap.get("project");
+    this.dataset = this._route.snapshot.paramMap.get("dataset");
+    this.table = this._route.snapshot.paramMap.get("table");
+    this.jobID = this._route.snapshot.paramMap.get("job");
+    this.location = this._route.snapshot.paramMap.get("location");
+    // To set the dropdown to "Auto Select", we need empty string for location.
+    if (!this.location) {
+      this.location = '';
+    }
+  }
```

Comment:
While it's verbose, I think I do prefer having these properties all set in ngOnInit.

Reply:
done.